Git hooks and github actions

brentmiller1973 opened the pull request:
I addressed the npm run test -- --changedSince=${{ github.head_ref }} issue (thanks @matthewroach ).  Not sure why the test is still failing however.  The git hooks are only hard coded to the base branch, spent too much time trying to figure out how to lint and test against base dynamically....I'm calling mercy.

Commit:
First commit - hooks

Commit:
Husky and actions - test

Commit:
Husky and actions - test

Commit:
Husky and actions - test

Commit:
Husky and actions - test

matthewroach commented on package.json:
```diff
@@ -6,8 +6,11 @@
   "scripts": {
     "test": "jest --coverage",
     "test:watch": "npm run test -- --watch -o --coverage",
+    "test:changed-feature-branch": "jest --changedSince=github.head_ref --coverage --passWithNoTests",
```
This is failing in the action as the value for `--changedSince` needs to be passed via the action like

```npm run test -- --changedSince=${{ github.head_ref }}```

matthewroach reviewed (changes requested):
The action is failing due to an issue with the `--changedSince` value which I added a comment too

Commit:
Updated workflow

matthewroach committed:
Update action to use base_ref

matthewroach committed:
Update comment

matthewroach committed:
fix merge issue

matthewroach commented:
@visionbegin - There was some issue with the Chromatic action I rebased from the base branch `arc-themes-release-version-1.19` - you'll need to pull from origin to get the latest updates

Regarding the changed since parameter - I've updated that to work - it needed to be `base_ref` not `head_ref` I made a mistake and got confused between the two keys and what values they provided.

Take a look over the PR and see if it all makes sense now?

brentmiller1973 commented:
> @visionbegin - There was some issue with the Chromatic action I rebased from the base branch `arc-themes-release-version-1.19` - you'll need to pull from origin to get the latest updates
> 
> Regarding the changed since parameter - I've updated that to work - it needed to be `base_ref` not `head_ref` I made a mistake and got confused between the two keys and what values they provided.
> 
> Take a look over the PR and see if it all makes sense now?

@matthewroach Thank you so much for the guidance!
